src/System.Linq.Expressions/tests/Unary/IncrementDecrementTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Linq.Expressions.Tests
+{
+    public abstract class IncrementDecrementTests
+    {
+        public struct Decrementable
+        {
+            public Decrementable(int value)
+            {
+                Value = value;
+            }
+
+            public int Value { get; }
+
+            public static Decrementable operator --(Decrementable operand) => new Decrementable(operand.Value - 1);
+        }
+
+        public struct Incrementable
+        {
+            public Incrementable(int value)
+            {
+                Value = value;
+            }
+
+            public int Value { get; }
+
+            public static Incrementable operator ++(Incrementable operand) => new Incrementable(operand.Value + 1);
```

Comment:
Comparing the differences in how it's done in C# and C++ and considering why might therefore be interesting, at least I found it so when C# came out. (As someone who at the time loved C++ but mostly wrote VB6 code for a living, when C# came out and was often being described as the next VB6 [the accuracy of that is another matter...] how the operator overloads compared with C++ was maybe the first thing I read up on).